ostree-backup.sh

```diff
@@ -24,44 +24,78 @@ if [[ -z "$backup_repo" ]]; then
 fi
 
 log_it "Saving list of running containers and clusterversion"
-mkdir /var/tmp/backup
-crictl ps -o json| jq -r '.containers[] | .imageRef' > /var/tmp/backup/containers.list
-oc get clusterversion version -ojson > /var/tmp/backup/clusterversion.json
+if [[ ! -f /tmp/container_list.done ]]; then
+    mkdir -p /var/tmp/backup
+    crictl ps -o json| jq -r '.containers[] | .imageRef' > /var/tmp/backup/containers.list
```

Comment:
This would probably need to be a list of all images, not just running containers, as there could be images run periodically that would be missed here:
crictl images -o json | jq -r '.images[].repoDigests[]'

Additionally, this should give us the complete list of images we'd need to precache and avoid the need for the "oc adm release extract" command in ostree-restore.sh - which I think likely gives us more images than we need on the SNO?

Reply:
Not to dismiss your points but my PR simply makes this code idempotent, I haven't changed it in any other regard.

Reply:
I haven't even fully understood the code / your points, perhaps it's best to discuss them somewhere else (preferably with @javipolo )

Reply:
instead of asking for `/tmp/container_list.done`, why not adjust the condition and simply ask for `/var/tmp/backup/clusterversion.json`? 

(we might avoid introducing a new empty file)

Reply:
If `oc get clusterversion version -ojson` runs into an error (e.g. when API happens to be down for whatever reason), `clusterversion.json` would still get created, but it would be empty

Reply:
This "done" pattern in scripts is pretty common even in the openshift installer: https://github.com/openshift/installer/blob/33a74c7d19a6264deeec1bef4f0f2fbb32c804b5/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L149

Reply:
As Omer says, this discussion belongs in a different place :) but let's keep it going hehe

> This would probably need to be a list of all images, not just running containers, as there could be images run periodically that would be missed here: crictl images -o json | jq -r '.images[].repoDigests[]'

100% agree. Will create a PR for that now

> Additionally, this should give us the complete list of images we'd need to precache and avoid the need for the "oc adm release extract" command in ostree-restore.sh - which I think likely gives us more images than we need on the SNO?

Let's talk about this with @eranco74 next week
I think my idea was to use the list of images that are currently running, and his idea was to use the images listed in `oc adm release extract`, and somehow we ending up merging both lists and pulling everything